app/agent/prebuild.service.test.ts

```diff
@@ -285,3 +294,148 @@ test(
     }
   }),
 );
+
+test.concurrent(
+  "checkoutAndInspect",
+  testEnv
+    .withBlockRegistry()
+    .withImagePush((timestamp) => [
+      {
+        tag: `quay.io/hocus/hocus-tests:checkout-and-inspect-${timestamp}`,
+        imageId: BlockRegistryService.genImageId("output"),
+      },
+    ])
+    .run(async ({ injector, runId }) => {
+      const prebuildService = injector.resolve(Token.PrebuildService);
+      const brService = injector.resolve(Token.BlockRegistryService);
+
+      const repoImageTag = "im1";
+      const repoImageId = await brService.loadImageFromRemoteRepo(
+        "quay.io/hocus/hocus-tests:fetchrepo-06-11-2023-18-13-34",
+        repoImageTag,
+      );
+      const repoContainerId = await brService.createContainer(repoImageId, "fetchrepo");
+      await expectContent(brService, {
+        numTotalContent: 2,
+      });
+      const runtime = injector.resolve(Token.QemuService)(runId);
+      const outputId = "output";
+      const tmpContentPrefix = "tmp-content-1337-";
+
+      await prebuildService.checkoutAndInspect({
+        runtime,
+        repoContainerId,
+        targetBranch: "checkout-and-inspect-test-1",
+        outputId,
+        projectConfigPaths: ["/"],
+        tmpContentPrefix,
+      });
+      await expectContent(brService, {
+        numTotalContent: 6,
+        prefix: {
+          value: tmpContentPrefix,
+          numPrefixedContent: 2,
+        },
+      });
+
+      const imId = await BlockRegistryService.genImageId(outputId);
+      const { mountPoint } = await brService.expose(imId, EXPOSE_METHOD.HOST_MOUNT);
+      expect(await doesFileExist(path.join(mountPoint, "project", "hocus.yml"))).toBe(true);
+    }),
+);
+
+test.concurrent(
+  "Test prebuild",
+  testEnv
+    .withBlockRegistry()
+    .withImagePush((timestamp) => [
+      {
+        tag: `quay.io/hocus/hocus-tests:prebuild-rootfs-${timestamp}`,
+        imageId: BlockRegistryService.genImageId("output-rootfs"),
+      },
+      {
+        tag: `quay.io/hocus/hocus-tests:prebuild-project-${timestamp}`,
+        imageId: BlockRegistryService.genImageId("output-project"),
+      },
+    ])
+    .run(async ({ injector, runId, db }) => {
+      const prebuildService = injector.resolve(Token.PrebuildService);
+      const brService = injector.resolve(Token.BlockRegistryService);
+
+      const [rootFsImageId, projectImageId] = await waitForPromises(
+        [
+          "quay.io/hocus/hocus-tests:buildfs-06-11-2023-18-32-43",
```

Comment:
Why is this timestamp hardcoded?

Reply:
That's the result of another test run that pushed this image. We have to hardcode the tag, otherwise we wouldn't know what to download to run this test.